crossplane/objects.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+from .parser import parse
+
+
+#
+# Functions
+#
+
+def _init_directive(directive_json):
+    return NGXBlockDirective(**directive_json) if 'block' in directive_json \
+        else NGXDirective(**directive_json)
+
+
+#
+# Classes
+#
+
+class NGXDirective(object):
+    __slots__ = ('directive', 'line', 'args', 'includes')
+
+    def __init__(self, directive='', line=0, args=[], includes=[], **kwargs):
+        self.directive = directive
+        self.line = line
+        self.args = args
+        self.includes = []
+
+    def dict(self):
+        return {
+            slot: getattr(self, slot)
+            for slot in self.__slots__
+        }
+
+
+class NGXBlockDirective(NGXDirective):
+    __slots__ = ('index', 'directive', 'line', 'args', 'block')
+
+    def __init__(self, block=[], **kwargs):
+        super(NGXBlockDirective, self).__init__(**kwargs)
+        self.index = {}
+        self.block = block
+
+        self._setup_block()
+
+    def _setup_block(self):
+        directives = []
+        for directive_json in self.block:
+            directives.append(_init_directive(directive_json))
+            self.__index(directives[-1])
+
+        self.block = directives
+
+    def __index(self, directive):
+        idx = self.index.get(directive.directive, [])
+        idx.append(directive)
+        self.index[directive.directive] = idx
+
+    def __contains__(self, item):
+        return item in self.index
+
+    def __getattr__(self, name):
```

Comment:
I'm not a fan of the \_\_getattr\_\_ idea from a class API perspective.

Reply:
Perhaps, as mentioned in the PR the API for working with these objects is something that could use a lot of polish.